fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/rules/PushFilterThroughFetchRule.java

```diff
@@ -236,11 +235,6 @@ private QueryPredicate pushLeafPredicate(@Nonnull RecordQueryFetchFromPartialRec
                                              @Nonnull CorrelationIdentifier oldInnerAlias,
                                              @Nonnull CorrelationIdentifier newInnerAlias,
                                              @Nonnull final QueryPredicate leafPredicate) {
-        if (leafPredicate instanceof QueryComponentPredicate) {
-            // We cannot push these predicates. They always contain nesteds.
-            return null;
-        }
```

Comment:
Is this change somehow related to serialization?

Reply:
I removed support for this kind of predicate which was long overdue. Didn't need serialization for it so rater removed it.